# U.S. Medical Insurance Costs

Purpose: The purpose of this project is to use python skills to organize the data in insurance.csv and to make it easier to use for analysis in the future. 

In [19]:
#import csv library
import csv

The first step of analyzing any data is to see what we are working with. For this I can use csv methods such as DictReader to see what data is available to me. 

From a quick peak at the data I can see we have 7 categories:
    1.) Age - string 
    2.) Sex - string
    3.) bmi - string
    4.) children - string
    5.) smoker - string
    6.) region - string
    7.) charges - string 

I also know that there is no missing data from this table. I will organize each column into its own lists which will make it easier to utilize and analyze. For columns that hold numerical data as strings, I will convert them to float values to make future computations easier 

In [20]:
age = []
sex = []
bmi = []
children = []
smoker = []
region = []
charges = []


with open('insurance.csv', newline = '') as insurance_info: 
    var2 = csv.DictReader(insurance_info)
    for row in var2:
        age.append(float(row['age']))
        sex.append(row['sex'])
        bmi.append(float(row['bmi']))
        children.append(float(row['children']))
        smoker.append(row['smoker'])
        region.append(row['region'])
        charges.append(float(row['charges']))


In [21]:
csv_length = len(age)

def average(list):
    length = len(list)
    total = 0
    for number in list:
        total += number
    average = total/length 
    print(average)


def find_options(list):
    variables = []
    for item in list: 
        if variables.count(item) == 0:
            variables.append(item)       
    return variables
    

def count(list):
    variables = find_options(list)
    statements = []
    for var in variables:
        total = []
        for item in list:
            if item == var:
                total.append(item)
        statements.append([var, len(total)])
    for pair in statements:
        print(str(pair[1]) + ' of ' + str(pair[0]))


sex_vs_charges =  list(zip(sex, charges))
region_vs_charges = list(zip(region, charges))
sex_vs_smoker = list(zip(sex, smoker))
age_vs_smoker = list(zip(age, smoker))

Now I can begin to use my functions and lists to draw conclusions about the entries in insurance.csv

In [22]:
average(age)

39.20702541106129


The average age of the entries in insurance.csv is about 39 years old. 

In [23]:
average(charges)

13270.422265141257


The average charge of the entries in insurance.csv is about $13,270. 

In [24]:
average(bmi)

30.663396860986538


The average bmi of the entries in insurance.csv is about 30. 

In [25]:
count(sex)
print(662/(csv_length))

662 of female
676 of male
0.4947683109118087


The distribution of males vs females in the insurance.csv dataset was about even with 49.47% of entries being female (662) and the rest male (676). 

In [26]:
count(region)
print('SW = ' + str(325/csv_length))
print('SE = ' + str(364/csv_length))
print('NW = ' + str(325/csv_length))
print('NE = ' + str(324/csv_length))

325 of southwest
364 of southeast
325 of northwest
324 of northeast
SW = 0.2428998505231689
SE = 0.27204783258594917
NW = 0.2428998505231689
NE = 0.242152466367713


The distribution of regions in the insurance.csv is roughly even with a slightly higher percentage (27%) of entries coming from the Southeast region compared to the Southwest, Northwest, and Northeast regions (24%). 

In [27]:
count(children)
print(str(574/csv_length))

574 of 0.0
324 of 1.0
157 of 3.0
240 of 2.0
18 of 5.0
25 of 4.0
0.4289985052316891


The distribution of the number of children is heavily skewed to the left with almost half of the entries coming from individuals with no children (43%). The category with the least number of entries would be those with 5 children (18) followed by 4 children (25), 3 children (157), 2 children (240), and 1 child (324). 

In [28]:
count(smoker)
print(str(1064/csv_length))

274 of yes
1064 of no
0.7952167414050823


The distribution of smokers vs non smokers is uneven with almost 80% of entries in insurance.csv counting themselves as individuals who do not smoke.

In [29]:
count(age)
average(age)
ages = sorted(age)
print(ages[0], ages[-1])
val = ages[0] + ((ages[-1] - ages[0])/2)
twenties = []
thirties = []
fourties= []
fifties = []
sixties = []
young = []
for age in ages:
    if age < 20:
        young.append(age)
    elif age<30:
        twenties.append(age)
    elif age<40:
        thirties.append(age)
    elif age<50:
        fourties.append(age)
    elif age<60: 
        fifties.append(age)
    else: 
        sixties.append(age)

print('Youngs:' + str(len(young)))
print('Twenties:' + str(len(twenties)))
print('Thirties:' + str(len(thirties)))
print('Fourties:' + str(len(fourties)))
print('Fifties:' + str(len(fifties)))
print('Sixties:' + str(len(sixties)))

68 of 19.0
69 of 18.0
28 of 28.0
26 of 33.0
26 of 32.0
27 of 31.0
29 of 46.0
25 of 37.0
23 of 60.0
28 of 25.0
23 of 62.0
28 of 23.0
26 of 56.0
28 of 27.0
29 of 52.0
27 of 30.0
26 of 34.0
25 of 59.0
23 of 63.0
26 of 55.0
28 of 22.0
28 of 26.0
25 of 35.0
28 of 24.0
27 of 41.0
25 of 38.0
25 of 36.0
28 of 21.0
29 of 48.0
27 of 40.0
25 of 58.0
28 of 53.0
27 of 43.0
22 of 64.0
29 of 20.0
23 of 61.0
27 of 44.0
26 of 57.0
27 of 29.0
29 of 45.0
28 of 54.0
28 of 49.0
29 of 47.0
29 of 51.0
27 of 42.0
29 of 50.0
25 of 39.0
39.20702541106129
18.0 64.0
Youngs:137
Twenties:280
Thirties:257
Fourties:279
Fifties:271
Sixties:114


The range of age entries in insurance.csv was from 18 yrs of age to 64 yrs of age. The age bracket with the most entries was those between the ages of 20 and 29 (280). There was a similar number of entries for those in their fourties and fifties. As expected, the age brackets of those under 20 or over 60 had the least number of entries. 

The next analyses will be done with my lists that combined two or more categories: sex_vs_charges, region_vs_charges, sex_vs_smoker, and age_vs_smoker. 

In [30]:
count(sex)
total = 0
for item in charges:
    total += item
print('total: ' + str(total))

total_fem = 0
total_male = 0

for item in sex_vs_charges:
    if item[0] == 'female':
        total_fem += item[1]
    if item[0] == 'male':
        total_male += item[1]
        
f_av = total_fem/662
m_av = total_male/676

print(f_av - m_av)
print(f_av)
print(m_av)

662 of female
676 of male
total: 17755824.990759
-1387.1723338865468
12569.57884383534
13956.751177721886


When comparing sex and charges, it appears that the average cost for males in insurance.csv is about $1,200 more expensive than that for females.

In [31]:
count(region)
SW = 0
SE = 0
NW = 0
NE = 0

for item in charges:
    total += item
print('total: ' + str(total))

for item in region_vs_charges:
    if item[0] == 'southwest':
        SW += item[1]
    elif item[0] == 'southeast':
        SE += item[1]
    elif item[0] == 'northwest':
        NW += item[1]
    else: 
        NE += item[1]

print('SW: ' + str(SW/325))
print('SE: ' + str(SE/364))
print('NW: ' + str(NW/325))
print('NE: ' + str(NE/324))


325 of southwest
364 of southeast
325 of northwest
324 of northeast
total: 35511649.98151798
SW: 12346.93737729231
SE: 14735.411437609895
NW: 12417.575373969228
NE: 13406.3845163858


When comparing region and charges, it appears as though the Southeast region has a higher charge cost on average. While southeast also has a higher number of entries, the average cost is still over 1,000 dollars more expensive than the next most expensive region (Northeast). It appears as though the Southwest and Northwest regions have similar average charges, with the Southwest region being the last expensive (12,000). 

In [32]:
count(sex)
count(smoker)
fem_yes = 0
fem_no = 0
m_yes = 0
m_no = 0


for item in sex_vs_smoker:
    if item[0] == 'female':
        if item[1] == 'yes':
            fem_yes += 1
        elif item[1] == 'no':
            fem_no += 1
    elif item[0] == 'male':
        if item[1] == 'yes':
            m_yes += 1
        elif item[1] == 'no':
            m_no +=1


f_yes_average = fem_yes / 662
f_no_average = fem_no / 662
m_yes_average = m_yes / 676
m_no_average = m_no / 676

f_pop_yes = fem_yes / len(sex)
f_pop_no = fem_no / len(sex)

m_pop_yes = m_yes / len(sex)
m_pop_no = m_yes / len(sex)


print(f_yes_average)
print(m_yes_average)

print(f_pop_yes)
print(m_pop_yes)

662 of female
676 of male
274 of yes
1064 of no
0.17371601208459214
0.23520710059171598
0.08594917787742899
0.11883408071748879


When comparing smoker status and sex, it appears that male entries are about 5% more likely to be yes than female entries. Additionally, smoking females represent less than 10% of the entries in insurance.csv. Females smokers represent about 17% of all female entries, and male smokers represent about 23% of all male entries. It appears that in either sex, the entry is more likely to be no on smoking than yes. 

In [33]:

smoker = []
non_smoker = []


for item in age_vs_smoker:
    if item[1] == 'yes':
        smoker.append(item[0])
    elif item[1] == 'no':
        non_smoker.append(item[0])

def find_age(list):
    twenties = []
    thirties = []
    fourties= []
    fifties = []
    sixties = []
    young = []

    for age in list:
        if age < 20:
            young.append(age)
        elif age<30:
            twenties.append(age)
        elif age<40:
            thirties.append(age)
        elif age<50:
            fourties.append(age)
        elif age<60: 
            fifties.append(age)
        else: 
            sixties.append(age)
    print(len(young))
    print(len(twenties))
    print(len(thirties))
    print(len(fourties))
    print(len(fifties))
    print(len(sixties))


smoker_sort = sorted(smoker)
non_sort = sorted(non_smoker)

print(smoker_sort[0], smoker_sort[-1])
print(non_sort[0], non_sort[-1])
print('smokers')
print(len(smoker))
find_age(smoker_sort)
print('nonsmokers')
print(len(non_smoker))
find_age(non_sort)


18.0 64.0
18.0 64.0
smokers
274
30
56
58
62
41
27
nonsmokers
1064
107
224
199
217
230
87


When comparing the group of non-smokers to smokers, it appears that while there are roughly 5 times as many nonsmokers as smokers, both groups have the same range of ages (18-64). The age group that appears to have the largest smoking proportion would be people in their 40s. 

The data is clean and ready to use for future analysis. While I did a few analyses on my own, there are many other calculations and conclusions to be drawn from this data set. I would be great to see more on how the number of children affects the charges, especially includng gender as a variable. For me, the next step with this data would be using R or other programs to create visual representations of the data. 